gateway-js/src/index.ts

```diff
@@ -912,7 +917,7 @@ export class ApolloGateway implements GraphQLService {
 
     // TODO(trevor:cloudconfig): This condition goes away completely
     if (
-      !this.experimental_schemaConfigDeliveryEndpoint &&
+      !this.schemaConfigDeliveryEndpoint &&
```

Comment:
should these be restructured as:

if (isPrecomposed)
else if (isLegacy)
else shouldn't-happen-error

perhaps?

(Mostly seems a bit concerning to see `!` used on a field which now has distinct behavior for null and undefined. I mean I think it's correct because of how isPrecomposedManagedConfig works but I don't see why we actually need this part of the condition at all?)

Reply:
Yeah this is a good change and becomes more readable through predicate usage. 👍 
0f67087